woocommerce/wc-admin
# Update table component to use wp card

Fixes (partially) #4002 

Updates the table and report table to use the new `Card` component.  Also updates and fixes up some styles in the table.

Styling here is nearly identical to the previous tables with the exception of table borders and shadows.  Note that the inner table border is now darker than the table outline, so I'm thinking we should probably update that to match the outer table border. /cc @elizaan36 @LevinMedia 

### Screenshots
<img width="787" alt="Screen Shot 2020-12-17 at 4 07 25 PM" src="https://user-images.githubusercontent.com/10561050/102544591-05f88a80-4083-11eb-9347-b9fcf0351c58.png">
<img width="754" alt="Screen Shot 2020-12-17 at 4 07 11 PM" src="https://user-images.githubusercontent.com/10561050/102544593-06912100-4083-11eb-88d3-e106646dfb34.png">
<img width="743" alt="Screen Shot 2020-12-17 at 4 07 03 PM" src="https://user-images.githubusercontent.com/10561050/102544594-06912100-4083-11eb-932f-98b5dc60110a.png">


### Detailed test instructions:

1. Create enough products to be paginated.
1. Navigate through the various reports.
2. Check that the table styles and actions/search/menu are styled correctly.
3. Also check various screen sizes.
